ghdocs/Contributing/NewComponent.md

```diff
@@ -0,0 +1,157 @@
+
+#### Fundamental components
+Fundamental components are the official React representation of our [Adobe XD Toolkit](https://static2.sharepointonline.com/files/fabric/fabric-website/files/officeuifabric_v4.0.0.zip) and receive higher priority in respect to stabilization, bug fixes, accessibility and general design resourcing. Components __cannot__ be immediately added as a Fundamental type component as it will need a period of stabilization and API review. All components should always start off as Experimental. A good portion of the Fundamental components are currently integrated within some of our major products such as OneDrive, SharePoint and Outlook. Fundamental components are developed in the office-ui-fabric-react project in the `<root of project>/packages/office-ui-fabric-react` folder.
+
+### Start coding your component
+
+#### Creating a component via the command line
+1. This is the recommended way of creating a new component.
+3. Open a command prompt in the root directory of your project directory.
+2. Run `npm run create-component -- --name ExcitingNewComponent`.
+
+#### Creating a component manually
+1. From the root of the project, navigate to `packages/experiments/src/components/`
+2. Create a new folder here with your component name
+  * Our components use the Pascal Case naming convention.
+  * In this example the component will be called `ExcitingNewComponent`.
+3. Create a `ExcitingNewComponent.types.ts` file that will contain an interface for your component props
+
+  * @todo: review
+  * Import React
+    * `import * as React from 'react'`;
+  * Import ExcitingNewComponent
+    * `import { ExcitingNewComponent } from './ExcitingNewComponent'`;
+    * Note: This class and file don't exist yet but they will during step 4 of this small tutorial.
+  * Create an empty interface `IExcitingNewComponent`
+    * `export interface IExcitingNewComponent {}`
+  * Add your props interface to this file.
+    * `export interface IExcitingNewComponentProps extends React.Props<ExcitingNewComponent> { … Props }`
+  * Create and export an interface for ExcitingNewComponent styles.
+
+```ts
+import * as React from 'react';
+import { ExcitingNewComponent } from './ExcitingNewComponent';
+import { IStyle } from '../../Styling';
+
+export interface IExcitingNewComponent {
+}
+
+export interface IExcitingNewComponentProps extends React.Props<ExcitingNewComponent> {
+  /**
+  * All props for your component are to be defined here.
+  */
+  componentRef?: (component: IExcitingNewComponent) => void;
```

Comment:
this should be moved up into the .types description, explaining this is how customers should access the imperative API surface (through `componentRef`)